pymatgen/core/structure.py

```diff
@@ -3980,67 +3984,174 @@ def set_charge(self, new_charge: float = 0.0) -> None:
 
     def relax(
         self,
-        calculator: str = "m3gnet",
+        calculator: str | Calculator = "m3gnet",
         relax_cell: bool = True,
+        optimizer: str = "FIRE",
```

Comment:
How about we make this `str | Optimizer` to match `calculator: str | Calculator`?